packages/@webex/internal-plugin-mercury/src/mercury.js

```diff
@@ -84,13 +84,17 @@ const Mercury = WebexPlugin.extend({
 
     this.connecting = true;
 
-    return Promise.resolve(
-      this.webex.internal.device.registered || this.webex.internal.device.register()
-    ).then(() => {
-      this.logger.info(`${this.namespace}: connecting`);
+    if (this.config.deviceRegistrationRequired ?? true) {
+      return Promise.resolve(
+        this.webex.internal.device.registered || this.webex.internal.device.register()
```

Comment:
@rsarika in a normal scenario before CC does mercury the device needs to be registred , shouldnt this line cover it and not not do actual registration 

If thats the case we just just have to make sure we check device is registred before mercury connection

Reply:
addressed and removed